ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetaDataAccessImpl.java

```diff
@@ -0,0 +1,156 @@
+package com.spectralogic.ds3client.metadata;
+
+import com.spectralogic.ds3client.helpers.Ds3ClientHelpers;
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Implementation of MetaDataAccess Interface
+ * Used to store meta data on Server
+ */
+public class MetaDataAccessImpl implements Ds3ClientHelpers.MetadataAccess {
+    private Map<String, Path> fileMapper = null;
+
+    public MetaDataAccessImpl(final Map<String, Path> fileMapper) {
+        this.fileMapper = fileMapper;
+    }
+
+    @Override
+    public Map<String, String> getMetadataValue(final String filename) {
+        Map<String, String> metadata = new HashMap<>();
+        try {
+            final Path file = fileMapper.get(filename);
+            metadata = storeMetaData(file);
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+        return metadata;
+    }
+
+    /*
+    * to store the meta data on server
+    *
+    * */
+    private Map<String, String> storeMetaData(final Path file) {
+        final Map<String, String> metadata = new HashMap<>();
```

Comment:
I would replace this with a Guava ImmutableMap.Builder<String, String> and then on line 93 return the built immutable map.